packages/perseus/src/widgets/interactive-graphs/graphs/angle.tsx

```diff
@@ -95,6 +100,97 @@ function AngleGraph(props: AngleGraphProps) {
+        endPoints,
+        centerPoint,
+        allowReflexAngles,
+    );
+    const angleMeasure = getWholeAngleMeasure(clockwiseCoords, centerPoint);
+    const angleLabel = `${allowReflexAngles ? 360 - angleMeasure : angleMeasure}`;
+
+    const {strings, locale} = usePerseusI18n();
+
+    const formatCoordinates = (x: number, y: number) => ({
+        x: srFormatNumber(x, locale),
+        y: srFormatNumber(y, locale),
+    });
+
+    const [initialSideAriaLabel, setInitialSideAriaLabel] =
+        React.useState<string>(
+            strings.srInitialSideAtCoordinates(
+                formatCoordinates(coords[2][X], coords[2][Y]),
+            ),
+        );
+
+    const [terminalSideAriaLabel, setTerminalSideAriaLabel] =
+        React.useState<string>(
+            strings.srTerminalSideAtCoordinates(
+                formatCoordinates(coords[0][X], coords[0][Y]),
+            ),
+        );
+
+    const [vertexAriaLabel, setVertexAriaLabel] = React.useState<string>(
+        strings.srVertexAtCoordinates(
+            formatCoordinates(coords[1][X], coords[1][Y]),
+        ),
+    );
+
+    const updateVertexAriaLabel = (newPoint: vec.Vector2) => {
+        const formattedVertexCoordinates = formatCoordinates(
+            newPoint[X],
+            newPoint[Y],
+        );
+        const label = showAngles
+            ? strings.srVertexWithAngleAtCoordinates({
+                  ...formattedVertexCoordinates,
+                  angle: angleLabel,
+              })
+            : strings.srVertexAtCoordinates(formattedVertexCoordinates);
+        setVertexAriaLabel(label);
+    };
+
+    const updateSideAriaLabel = (
+        side: "terminal" | "initial",
+        newPoint: vec.Vector2,
+    ): void => {
+        const formattedCoordinates = formatCoordinates(
+            newPoint[X],
+            newPoint[Y],
+        );
+        side === "terminal"
+            ? setTerminalSideAriaLabel(
+                  strings.srUpdatedTerminalSideAtCoordinates(
+                      formattedCoordinates,
+                  ),
+              )
+            : setInitialSideAriaLabel(
+                  strings.srUpdatedInitialSideAtCoordinates(
+                      formattedCoordinates,
+                  ),
+              );
+    };
+
+    const handleSidePointMove = (
+        coordIndex: number,
+        newPoint: vec.Vector2,
+        side: "terminal" | "initial",
+    ): unknown => {
+        updateVertexAriaLabel(coords[1]);
+        updateSideAriaLabel(side, newPoint);
+        return dispatch(actions.angle.movePoint(coordIndex, newPoint));
+    };
+
+    const handleVertexMove = (coordIndex: number, newPoint: vec.Vector2) => {
+        updateVertexAriaLabel(newPoint);
+        return dispatch(actions.angle.movePoint(coordIndex, newPoint));
+    };
+
+    React.useEffect(() => {
+        updateVertexAriaLabel(coords[1]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [showAngles]);
+
```

Comment:
I think we could avoid using the `useState` and `useEffect` hooks here if we:

- compute aria labels from the `graphState` prop
- use aria-live to notify the user of changes to the graph

This would ensure that the SR view and the visible view never get out of sync, since both would be pure functions of `graphState`. Currently, all the individual graph components are stateless (or nearly so — there are a couple `useState`s for hover and dragging effects) and I'd like to keep it that way.

Wonder Blocks is [planning an API for sending aria-live notifications](https://khanacademy.atlassian.net/wiki/spaces/WB/pages/3418095646/Announcer+Implementation+Spec), but since it's not built yet, I think what we could do is write a minimal implementation of the same API and swap in the WB one when it's ready. I'd be down to pair or consult on that if you want.